src/main/java/uk/gov/ch/service/update/impl/ManagingOfficerDataServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.ch.service.update.impl;
+
+import uk.gov.ch.service.update.ManagingOfficerDataService;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import uk.gov.ch.OracleQueryApplication;
+import uk.gov.ch.exception.ManagingOfficerCountNotFoundException;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+import uk.gov.ch.model.update.ManagingOfficerData;
+import uk.gov.ch.repository.update.ManagingOfficerDataRepository;
+
+@Service
+public class ManagingOfficerDataServiceImpl implements ManagingOfficerDataService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(OracleQueryApplication.APPLICATION_NAME_SPACE);
+
+    private static final String NO_MANAGING_OFFICERS_FOUND_MESSAGE = "No managing officers were found.";
+
+    @Autowired
+    private ManagingOfficerDataRepository managingOfficerDataRepository;
+
+    @Override
+    public List<ManagingOfficerData> getManagingOfficerData(String companyNumber, Pageable pageable)
+            throws ManagingOfficerCountNotFoundException {
+        List<ManagingOfficerData> list = managingOfficerDataRepository
+                .getManagingOfficerData(companyNumber, pageable).toList();
+        LOGGER.debug(String.format("getManagingOfficerData Query list size returned = %s", list.size()));
+        if (list.isEmpty()) {
+            LOGGER.info(NO_MANAGING_OFFICERS_FOUND_MESSAGE);
+            throw new ManagingOfficerCountNotFoundException(NO_MANAGING_OFFICERS_FOUND_MESSAGE);
```

Comment:
Do we want to throw an exception even if there are no MOs found?

Reply:
i think so yes, if there are no MO's it should return a 404 or it could return an empty list 🤷